### ISSUE TITLE
Dropping GIL when calling Julia from Python (JuliaCall)

### ISSUE THREAD

**Issue by Moelf** [JuliaPy/PythonCall.jl#343]
Similar to what Numba's `nogil`, it's useful for using Julia as a performant backend of Python

**Comment by cjdoris** [comment]
Repeating what I said in #108

> Yeah calling into Julia from Julia doesn't unlock the GIL. This is because the Julia code could well call back into Python, which would require automatically relocking the GIL, which is all quite complicated and probably slow in general. If you can guarantee your Julia code doesn't call back into Python, you can release the GIL yourself. A future version of JuliaCall will allow something like some_julia_function.jl_call_nogil(x, y, z) to make this a bit easier.

Specifically, you can release the GIL yourself in the Julia code you call like:
```julia
pythread = PythonCall.C.PyEval_SaveThread()
try
    # code which doesn't touch Python
finally
    PythonCall.C.PyEval_RestoreThread(pythread)
end
```

**Comment by PallHaraldsson** [comment]
In case relevant, new in 3.12rc1:
https://peps.python.org/pep-0684/

https://docs.python.org/3.12/whatsnew/3.12.html#pep-684-a-per-interpreter-gil
>PEP 684: A Per-Interpreter GIL
>Sub-interpreters may now be created with a unique GIL per interpreter. This allows Python programs to take full advantage of multiple CPU cores.
See also new in 3.13:

>PEP 554 – Multiple Interpreters in the Stdlib

>**A Disclaimer about the GIL**
>To avoid any confusion up front: This PEP is meant to be independent of any efforts to stop sharing the GIL between interpreters (PEP 684). At most this proposal will allow users to take advantage of any GIL-related work.

More detailed info at: https://github.com/JuliaPy/PyCall.jl/issues/1046#issuecomment-1697190563

**Comment by Moelf** [comment]
is this closed by #535?

**Comment by cjdoris** [comment]
Yep thanks.
